pcluster/pcluster_stack.py

```diff
@@ -134,24 +167,77 @@ def __init__(self, scope: cdk.Construct, id: str, **kwargs) -> None:
              secret_string=c9_createkeypair_cr.get_att_string('PrivateKey')
         )
 
-        # The iam policy has a <REGION> parameter that needs to be replaced.
+        # The iam policy has parameters like <REGION> that need to be replaced.
         # We do it programmatically so future versions of the synth'd stack
         # template include all regions.
+        key_vals={
+            '<AWS ACCOUNT ID>': self.account,
+            '<PARALLELCLUSTER EC2 ROLE NAME>': 'parallecluster-*',
+            '<REGION>': [r.name for r in region_info.RegionInfo.regions],
+            '<RESOURCES S3 BUCKET>': [data_bucket.bucket_name, 'parallelcluster-*']
+        }
+        parallelcluster_user_policy=[]
         with open('iam/ParallelClusterUserPolicy.json') as json_file:
             data = json.load(json_file)
-            for s in data['Statement']:
-                if s['Sid'] == 'S3ParallelClusterReadOnly':
-                    s['Resource'] = []
-                    for r in region_info.RegionInfo.regions:
-                        s['Resource'].append('arn:aws:s3:::{0}-aws-parallelcluster*'.format(r.name))
 
-            parallelcluster_user_policy = iam.CfnManagedPolicy(self, 'ParallelClusterUserPolicy', policy_document=iam.PolicyDocument.from_json(data))
+            # Inner def replaces strings and expands lists
```

Comment:
Whoa! this got really complicated really fast. Can we replace this with anything simpler? Seems like all this section does is create the IAM policy correct? what if we generalized the IAM polciy and removed region specific arns?

Reply:
I'm still really skeptical of this entire section. I acknowledge that this keeps the IAM policy separate and allows us to sub in parameters and might be the best way to do it but I still think it's entirely too complicated. I'd like to see something similar to the way we do it in post-install script